Editor/EditorCore/EditorToolsContexts.cs

```diff
@@ -8,6 +17,348 @@ class PositionToolContext : EditorToolContext
+                    ? DropdownMenuAction.Status.Checked
+                    : DropdownMenuAction.Status.Normal);
+            menu.AppendHeaderAction(UI.EditorGUIUtility.Styles.FaceIcon,
+                x => { ProBuilderEditor.selectMode = SelectMode.Face; },
+                x => ProBuilderEditor.selectMode == SelectMode.Face
+                    ? DropdownMenuAction.Status.Checked
+                    : DropdownMenuAction.Status.Normal);
+
+            var actions = EditorToolbarLoader.GetActions();
+            var group = ToolbarGroup.Tool;
+
+            // grouping and filtering is bespoke for demo reasons
+            foreach (var action in actions)
+            {
+                if (k_ContextMenuBlacklist.Contains(action.GetType()))
+                    continue;
+
+                if (action.group == ToolbarGroup.Entity || action.group == ToolbarGroup.Object)
+                    continue;
+
+                if (action.group != group)
+                {
+                    menu.AppendSeparator();
+                    group = action.group;
+                }
+
+                var title = action.menuTitle;
+                if (action.group != ToolbarGroup.Geometry && action.group != ToolbarGroup.Tool)
+                    title = $"{action.group}/{action.menuTitle}";
+
+                if (action.hasOptions)
+                    menu.AppendAction(title, _ => EditorAction.Start(new MenuActionSettings(action)), GetStatus(action), action.icon);
+                else
+                    menu.AppendAction(title, _ => action.PerformAction(), GetStatus(action), action.icon);
+            }
+
+            var trs = Selection.transforms;
+            if (trs.GetComponents<MeshFilter>().Length > trs.GetComponents<ProBuilderMesh>().Length)
+            {
+                menu.AppendSeparator();
+                ContextMenuUtility.AddMenuItemsForType(menu, typeof(MeshFilter), targets, "Mesh Filter");
+            }
+        }
+
+        // This boolean allows to call the action only once in case of multi-selection as PB actions
+        // are called on the entire selection and not per element.
+        static bool s_ActionAlreadyTriggered = false;
+
+        [MenuItem("CONTEXT/ProBuilderMesh/Conform Normals", true)]
+        static bool ValidateConformObjectNormalsAction()
+        {
+            return MeshSelection.selectedObjectCount > 0;
+        }
+
+#if UNITY_2023_3_OR_NEWER
+        [MenuItem("CONTEXT/ProBuilderMesh/Conform Normals", false, 11, "Conform object normals","Packages/com.unity.probuilder/Content/Icons/Toolbar/Object_ConformNormals@2x.png")]
+#else
+        [MenuItem("CONTEXT/ProBuilderMesh/Conform Normals", false, 11)]
+#endif
+        static void ConformObjectNormalsAction(MenuCommand command)
+        {
+            if (!s_ActionAlreadyTriggered)
+            {
+                s_ActionAlreadyTriggered = true;
+                //Once again, delayCall is necessary to prevent multiple call in case of multi-selection
```

Comment:
This is necessary because context menu callbacks are invoked for each object in the selection, and `MenuAction.PerformAction` expects to be called once, right?